app/scripts/services/DataRepositoryFactory.js

```diff
@@ -22,7 +24,7 @@ var DataRepositoryFactory = function ($firebaseObject, $firebaseArray) {
     var self = this;
 
     this.syncObject.$loaded().then(function () {
-      self.dataRepository = new DataRepository(self.recordSets);
+      self.dataRepository = new DataRepository(self.recordSets, new AuditLogger(self.recordSets));
```

Comment:
I thought about what you were saying, and came to the conclusion that you were right!
When I originally wrote the code I had to decide what was responsible for deciding which audit logger to use (in theory there could be many), and I decided that the DataRepositoryFactory should decide, which is why I did it the way I did (or maybe I just did it like that because the DataRepositoryFactory had the recordSets? I have such a terrible memory!). However, after what you said, I realised we can't pass any old audit logger in, because it has to be consistent with the records that the DataRepository is dealing with (I hope that makes sense), and that means the DataRepository should actually instantiate the audit logger - so in fact, I've gone further than what you suggested, and now it doesn't pass in an audit logger at all, but instantiates it on line 14.
Of course it just happens that the audit logger as it's written doesn't know (and doesn't need to know) anything about the structure of the order records (brands, categories and products), but the point is that the DataRepository needs to make the decision as to which logger to use. Well, I _think_ that's right anyway!!